sql/scan.go

```diff
@@ -337,7 +337,16 @@ func (n *scanNode) initWhere(where *parser.Where) error {
 	if n.err == nil {
 		// Normalize the expression (this will also evaluate any branches that are
 		// constant).
-		n.filter, n.err = parser.NormalizeAndTypeCheckExpr(n.filter)
+		n.filter, n.err = parser.NormalizeExpr(n.filter)
+	}
+	var whereType parser.Datum
+	if n.err == nil {
+		whereType, n.err = parser.TypeCheckExpr(n.filter)
+	}
+	if n.err == nil {
```

Comment:
Per my other comment, I think you might need to allow `whereType == parser.DNull` as well.